src/import/FSHImporter.ts

```diff
@@ -1380,8 +1421,110 @@ export class FSHImporter extends FSHVisitor {
     const insertRule = new InsertRule()
       .withLocation(this.extractStartStop(ctx))
       .withFile(this.currentFile);
-    insertRule.ruleSet = ctx.SEQUENCE().getText();
-    return insertRule;
+    const [rulesetName, ruleParams] = this.parseRulesetReference(ctx.RULESET_REFERENCE().getText());
+    insertRule.ruleSet = rulesetName;
+    if (ruleParams) {
+      insertRule.params = this.parseInsertRuleParams(ruleParams);
+      const ruleSet = this.paramRuleSets.get(insertRule.ruleSet);
+      if (ruleSet) {
+        const ruleSetIdentifier = Immutable.List<string>([ruleSet.name, ...insertRule.params]);
+        if (
+          ruleSet.parameters.length === insertRule.params.length &&
+          !this.currentDoc.appliedRuleSets.has(ruleSetIdentifier)
```

Comment:
If the `currentDoc` does have the given `ruleSetIdentifier` already, that drops into the `Incorrect number of parameters` error condition (unless I'm getting mixed up in my nesting here), and that doesn't seem to make sense.

Should this be an error condition in the first place? I'm not sure why a certain rule set couldn't be inserted twice into the same document with the same params.

Reply:
Ah, I shouldn't have combined both of those boolean expressions. It is, of course, completely fine to insert a rule set twice with the same parameters. There's no need to generate the applied rule set again, so that step can be skipped, but nothing wrong has occurred.